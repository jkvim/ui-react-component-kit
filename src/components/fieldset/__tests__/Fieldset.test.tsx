import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Fieldset } from '../';

describe('Fieldset', () => {

  describe('props', () => {

    const wrapper = mount(
      <Fieldset id="id" legend="legend" className="class" />);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `id` in props', () => {
      expect(wrapper.props().id).toBe('id');
    });

    it('should pass `legend` in props', () => {
      expect(wrapper.props().legend).toBe('legend');
    });

    it('should pass `className` in props', () => {
      expect(wrapper.props().className).toBe('class');
    });
  });
});
