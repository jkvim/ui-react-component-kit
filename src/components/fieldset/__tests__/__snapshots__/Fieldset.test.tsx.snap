// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset props should match snapshot 1`] = `
<Fieldset
  className="class"
  id="id"
  legend="legend"
>
  <styled.fieldset
    className="class"
    id="id"
  >
    <fieldset
      className="class sc-bdVaJa bDWFJH"
      id="id"
    >
      <legend>
        legend
      </legend>
    </fieldset>
  </styled.fieldset>
</Fieldset>
`;
